protocol/src/main/java/org/geysermc/mcprotocollib/network/Session.java

```diff
@@ -183,26 +184,21 @@ public interface Session {
     void callPacketSent(Packet packet);
 
     /**
-     * Gets the compression packet length threshold for this session (-1 = disabled).
+     * Sets the compression config for this session.
      *
-     * @return This session's compression threshold.
+     * @param compressionConfig the compression to compress with,
+     *                          or null to disable compression
      */
-    int getCompressionThreshold();
+    void setCompression(CompressionConfig compressionConfig);
```

Comment:
nullable annotation, same w encryption setter, and apply to TcpSession as well

Reply:
done